clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -354,6 +358,467 @@ fatbinary(ArrayRef<std::pair<StringRef, StringRef>> InputFiles,
+static Expected<StringRef> convertSPIRVToIR(StringRef Filename,
+                                            const ArgList &Args) {
+  Expected<std::string> SPIRVToIRWrapperPath = findProgram(
+      "spirv-to-ir-wrapper", {getMainExecutable("spirv-to-ir-wrapper")});
+  if (!SPIRVToIRWrapperPath)
+    return SPIRVToIRWrapperPath.takeError();
+
+  // Create a new file to write the converted file to.
+  auto TempFileOrErr =
+      createOutputFile(sys::path::filename(ExecutableName), "bc");
+  if (!TempFileOrErr)
+    return TempFileOrErr.takeError();
+
+  BumpPtrAllocator Alloc;
+  StringSaver Saver(Alloc);
+
+  SmallVector<StringRef, 16> CmdArgs;
+  CmdArgs.push_back(*SPIRVToIRWrapperPath);
+  CmdArgs.push_back(Filename);
+  CmdArgs.push_back("-o");
+  CmdArgs.push_back(*TempFileOrErr);
+  CmdArgs.push_back("-llvm-spirv-opts");
+  CmdArgs.push_back("\"--spirv-preserve-auxdata");
+  CmdArgs.push_back("--spirv-target-env=SPV-IR");
+  CmdArgs.push_back("--spirv-builtin-format=global\"");
+  if (Error Err = executeCommands(*SPIRVToIRWrapperPath, CmdArgs))
+    return std::move(Err);
+  return *TempFileOrErr;
+}
+
+// Run sycl-post-link tool
+static Expected<StringRef>
+runSYCLPostLink(SmallVector<StringRef, 16> &InputFiles, const ArgList &Args) {
+  Expected<std::string> SYCLPostLinkPath =
+      findProgram("sycl-post-link", {getMainExecutable("sycl-post-link")});
+  if (!SYCLPostLinkPath)
+    return SYCLPostLinkPath.takeError();
+
+  // Create a new file to write the output of sycl-post-link to.
+  auto TempFileOrErr =
+      createOutputFile(sys::path::filename(ExecutableName), "table");
+  if (!TempFileOrErr)
+    return TempFileOrErr.takeError();
+
+  BumpPtrAllocator Alloc;
+  StringSaver Saver(Alloc);
+
+  StringRef SYCLPostLinkOptions;
+  if (Arg *A = Args.getLastArg(OPT_sycl_post_link_options_EQ))
+    SYCLPostLinkOptions = A->getValue();
+
+  SmallVector<StringRef, 16> CmdArgs;
+  CmdArgs.push_back(*SYCLPostLinkPath);
+  while (!SYCLPostLinkOptions.empty()) {
+    auto [Option, Rem] = SYCLPostLinkOptions.split(" ");
+    CmdArgs.push_back(Option);
+    SYCLPostLinkOptions = Rem;
+  }
+  CmdArgs.push_back("-o");
+  CmdArgs.push_back(*TempFileOrErr);
+  for (auto &InputFile : InputFiles)
+    CmdArgs.push_back(InputFile);
+  if (Error Err = executeCommands(*SYCLPostLinkPath, CmdArgs))
+    return std::move(Err);
+  return *TempFileOrErr;
+}
+
+// This table is used to manage the output table populated by sycl-post-link.
+struct Table {
+  typedef struct {
+    std::string IRFile;
+    std::string PropFile;
+    std::string SymFile;
+  } SYCLTableEntry;
+
+  SmallVector<SYCLTableEntry, 16> Entries;
+
+  SmallVector<std::string, 16> getListOfIRFiles(void) {
+    SmallVector<std::string, 16> Files;
+    for (auto Entry : Entries) {
+      Files.push_back(Entry.IRFile);
+    }
+    return Files;
+  }
+
+  void printSYCLTable(void) {
+    llvm::errs() << "Print SYCL Table\n";
```

Comment:
Why do we print to llvm::errs() stream? I'm not sure what is the purpose of this method as I can't find uses for it. Please, bear in mind that LIT considers tests writing to err stream as failed.

Reply:
I was using this routine for localized debugging. I can remove this.

Thanks